lib/semantics/tools.cc

```diff
@@ -387,4 +385,33 @@ const Symbol *FindUltimateComponent(const DerivedTypeSpec &derivedTypeSpec,
   return nullptr;
 }
 
+bool IsFinalizable(const Symbol &symbol) {
+  const DeclTypeSpec *type{symbol.GetType()};
+  if (type) {
+    const DerivedTypeSpec *derived{type->AsDerived()};
+    if (derived) {
+      const Scope *scope{derived->scope()};
+      if (scope) {
```

Comment:
`type`, `derived`, and `scope`, same as above.

Reply:
Thanks.  I'll change that.